aip/general/0162.md

```diff
@@ -27,94 +27,66 @@ to the version of an API as a whole.
 
 APIs **may** store a revision history for a resource if it is useful to users.
 
-APIs implementing resources with a revision history **must** provide a
-`revision_id` field on the resource:
+APIs implementing resources with a revision history **should** abstract resource revisions as nested collection of the resource. The
+resource revision type **must** inherit from the parent resource and has a field being the parent resource type.
 
 ```proto
-message Book {
-  // The name of the book.
-  string name = 1;
+message BookRevision {
+   // The name of the book revision.
+   string name = 1;
 
-  // Other fields…
-
-  // The revision ID of the book.
-  // A new revision is committed whenever the book is changed in any way.
-  // The format is an 8-character hexadecimal string.
-  string revision_id = 5 [
-    (google.api.field_behavior) = IMMUTABLE,
-    (google.api.field_behavior) = OUTPUT_ONLY];
+   // The snapshot of the book
+   Book book = 2 [(google.api.field_behavior) = IMMUTABLE];
 
   // The timestamp that the revision was created.
-  google.protobuf.Timestamp revision_create_time = 6
+  google.protobuf.Timestamp create_time = 3
     [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // tags set on the resource revisions by service or user.
+  repeated string tags = 4;
+  [(google.api.field_behavior) = OUTPUT_ONLY];
 }
 ```
 
-- The resource **must** contain a `revision_id` field, which **should** be a
-  string and contain a short, automatically-generated random string. A good
-  rule of thumb is the last eight characters of a [UUID version 4 (random)](UUID4).
-  - The `revision_id` field **must** document when new revisions are created
-    (see [committing revisions](#committing-revisions) below).
-  - The `revision_id` field **should** document the format of revision IDs.
-- The resource **must** contain a `revision_create_time` field, which
-  **should** be a `google.protobuf.Timestamp` (see [AIP-142][]).
+- The resource revision **must** contain a field representing the snapshot of the parent resource.
+- The resource revision **must** contain a `create_time` field, which
+  **must** be a `google.protobuf.Timestamp` (see [AIP-142][]).
```

Comment:
```suggestion
- The resource revision **must** contain a `create_time` field (see [AIP-142][]).
```

I'm thinking this is better because we don't duplicate guidance on the type of create time - we have a few AIPs that are hard to keep in sync because of duplicated guidance.